fog/ledger/server/src/sharding_strategy.rs

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Enables a Key Image Store to know for which blocks to process key images.
+//!
+//! By determining which key images to process, we are able to "shard" the set
+//! of key images across Key Image Store instances.
+
+use mc_blockchain_types::BlockIndex;
+use mc_fog_types::{common::BlockRange, BlockCount};
+use serde::Serialize;
+use std::str::FromStr;
+
+/// Tells a Key Image Store for which blocks it should process key images.
+pub trait ShardingStrategy {
+    /// Returns true if the Key Image Store should process this block.
+    fn should_process_block(&self, block_index: BlockIndex) -> bool;
+
+    /// Returns true if the Key Image Store is ready to serve key images to the
+    /// client.
+    ///
+    /// Different sharding strategies might be ready to serve key images when
+    /// different conditions have been met.
+    fn ready(&self, processed_block_count: BlockCount) -> bool;
+
+    /// Returns the block range that this sharding strategy is responsible for.
+    fn get_block_range(&self) -> BlockRange;
+}
+
+/// Determines whether or not to process a block's key images based on the
+/// "epoch" sharding strategy, in which a block is processed IFF it falls within
+/// the contiguous range of blocks.
+///
+/// In practice, the set of Key Image Shards will contain overlapping
+/// [epoch_block_ranges] in order to obfuscate which shard processed the key
+/// images.
+#[derive(Clone, Serialize)]
+pub struct EpochShardingStrategy {
+    /// If a block falls within this range, then the Key Image Store should
+    /// process its key images.
+    epoch_block_range: BlockRange,
+}
+
+impl ShardingStrategy for EpochShardingStrategy {
+    fn should_process_block(&self, block_index: BlockIndex) -> bool {
+        self.epoch_block_range.contains(block_index)
+    }
+
+    fn ready(&self, processed_block_count: BlockCount) -> bool {
+        self.have_enough_blocks_been_processed(processed_block_count)
+    }
+
+    fn get_block_range(&self) -> BlockRange {
+        self.epoch_block_range.clone()
+    }
+}
+
+impl Default for EpochShardingStrategy {
+    fn default() -> Self {
+        Self {
+            epoch_block_range: BlockRange::new(0, u64::MAX),
+        }
+    }
+}
+
+impl EpochShardingStrategy {
+    #[allow(dead_code)]
```

Comment:
Disabling a lint/compiler warning should be avoided if possible, explained if not.
```suggestion
```

Reply:
If this only exists to satisfy unit tests, consider `#[cfg(test)]`, or just having a helper function in the test module.
```suggestion
    #[cfg(test)]
```